compiler/rustc_mir/src/transform/coverage/counters.rs

```diff
@@ -95,6 +105,7 @@ impl CoverageCounters {
     /// Counter IDs start from one and go up.
     fn next_counter(&mut self) -> CounterValueReference {
         assert!(self.next_counter_id < u32::MAX - self.num_expressions);
+        assert!(self.next_counter_id <= MAX_COUNTER_GUARD);
```

Comment:
You have `+ 1` above and `<=` here. I guess it doesn't really matter but I'd expect one or the other, not both.

Reply:
Agreed, and at first I did remove the `+ 1` from the constant def, but (as I will explain below) I believe I'm able to remove `MAX_COUNTER_GUARD` completely, which is even better. So that second `assert!` is gone as well.